winter/core/annotation_decorator.py

```diff
@@ -0,0 +1,52 @@
+import inspect
+import types
+import typing
+
+from .component import Component
+from .component_method import ComponentMethod
+
+
+def annotate(
+        value: typing.Any,
+        func_or_cls: typing.Optional[typing.Union[types.FunctionType, ComponentMethod, typing.Type]] = None
+) -> typing.Callable:
+    if func_or_cls is None:
+        return lambda func_or_cls: annotate(value, func_or_cls)
+
+    if isinstance(func_or_cls, ComponentMethod) or isinstance(func_or_cls, types.FunctionType):
+        return annotate_method(value, func_or_cls)
+    elif inspect.isclass(func_or_cls):
+        return annotate_class(value, func_or_cls)
+    else:
+        raise ValueError(f'Need function or class. Got: {func_or_cls}')
+
+
+def annotate_class(
+        value: typing.Any,
+        cls: typing.Optional[ComponentMethod] = None
```

Comment:
Why None by default?

Reply:
For being able to do somethin like this

@annotate_class(Route) and annotate_class(Route('test), SimpleComponent
As decorator and as just function

Reply:
Let's left a single API without the parameter, just to make it straightforward. If it's a decorator then let it be the decorator only. In a rare case when you want to test it, then just call it `annotate_class(Route('test'))(SimpleComponent)`.